controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1514,158 +1498,88 @@ int bml_internal::set_wifi_credentials(const std::string ssid, const std::string
-            beerocks_message::cACTION_PLATFORM_BEEROCKS_CREDENTIALS_UPDATE_REQUEST>(cmdu_tx);
-
-        if (config == nullptr) {
-            LOG(ERROR)
-                << "Failed building ACTION_PLATFORM_BEEROCKS_CREDENTIALS_UPDATE_REQUEST message!";
-            // Clear the promise holder
-            m_prmWiFiCredentialsUpdate = nullptr;
-            return (-BML_RET_OP_FAILED);
-        }
-
-        string_utils::copy_string(config->params().ssid, ssid.c_str(),
-                                  message::WIFI_SSID_MAX_LENGTH);
-        string_utils::copy_string(config->params().pass, pass.c_str(),
-                                  message::WIFI_PASS_MAX_LENGTH);
-
-        switch (sec) {
-        case BML_WLAN_SEC_NONE:
-            config->params().sec = beerocks_message::eWiFiSec_None;
-            break;
-
-        case BML_WLAN_SEC_WEP64:
-            config->params().sec = beerocks_message::eWiFiSec_WEP64;
-            break;
-
-        case BML_WLAN_SEC_WEP128:
-            config->params().sec = beerocks_message::eWiFiSec_WEP128;
-            break;
-
-        case BML_WLAN_SEC_WPA_PSK:
-            config->params().sec = beerocks_message::eWiFiSec_WPA_PSK;
-            break;
-
-        case BML_WLAN_SEC_WPA2_PSK:
-            config->params().sec = beerocks_message::eWiFiSec_WPA2_PSK;
-            break;
+    // Send to master for credential update
+    LOG(TRACE) << "BML:: Config update prepareing message to send to master";
 
-        case BML_WLAN_SEC_WPA_WPA2_PSK:
-            config->params().sec = beerocks_message::eWiFiSec_WPA_WPA2_PSK;
-            break;
+    // Increase the operation timeout to 3 minutes (for global update)
+    iOpTimeout = 180000;
 
-        default: {
-            LOG(WARNING) << "Unsupported Wi-Fi security: " << sec;
-            // Clear the promise holder
-            m_prmWiFiCredentialsUpdate = nullptr;
-            return (false);
-        }
-        }
-
-        // Set the rest of the message values
-        config->params().vap_id = uint8_t(vap_id);
-        config->params().force  = (force) ? 1 : 0;
-
-        // Build and send the message
-
-        if (!message_com::send_cmdu(m_sockPlatform, cmdu_tx)) {
-            LOG(ERROR) << "Failed sending configuration update message!";
-            // Clear the promise holder
-            m_prmWiFiCredentialsUpdate = nullptr;
-            return (-BML_RET_OP_FAILED);
-        }
-
-        // Forward credentials update request to master
-    } else {
-
-        // Send to master for credential update
-        LOG(TRACE) << "BML:: Config update prepareing message to send to master";
-
-        // Increase the operation timeout to 3 minutes (for global update)
-        iOpTimeout = 180000;
+    // Command supported only on local master
+    if (!is_local_master()) {
+        LOG(ERROR) << "Command supported only on local master!";
+        // Clear the promise holder
+        m_prmWiFiCredentialsUpdate = nullptr;
```

Comment:
I didn't change it. Only remove the part that sends this message directly to the agent. For some reason, the git diff looks like I changed it. Anyway, this should not be removed since it is updating the controller about new credentials, which will eventually trigger auto-config.